### PR TITLE
Fix nav

### PR DESCRIPTION
**Pull request by JustinW1395**
Fixed all the issues with the main app navigation.

### CODE REVIEW

File: report_my_wreck/report_my_wreck/MapView.swift [hunk 1]
```diff
@@ -8,20 +8,20 @@
...
     var body: some View {
         Text("maps incoming")
-        NavigationView {
-            NavigationLink(destination: EnterDetailsView()) {
-                Text("Next Page")
-            }
+        NavigationLink(destination: EnterDetailsView(shouldPopToRootView: self.$rootIsActive)) {
+            Text("Next Page")
         }
+        .isDetailLink(false)
```
Comment (RJB888) [inline report_my_wreck/report_my_wreck/MapView.swift:17]: What does this do, and why is it only here?

File: report_my_wreck/report_my_wreck/EnterDetailsView.swift [hunk 2]
```diff
@@ -27,14 +22,16 @@ struct EnterDetailsView: View {
                     Text("Select Photo")
                 }
             }
-            
+            Button (action: { self.shouldPopToRootView = false } ){
```
Comment (RJB888) [inline report_my_wreck/report_my_wreck/EnterDetailsView.swift:25]: How will this work when we actually have "actions" for the buttons to do?

**Review by RJB888 — CHANGES_REQUESTED** [review]

File: report_my_wreck/report_my_wreck/EnterDetailsView.swift (hunk 2 above)
Comment (JustinW1395) [reply report_my_wreck/report_my_wreck/EnterDetailsView.swift:25]: Well this button just takes you back to the main screen and closes out the navigationview. We can still add other buttons, it's just this one ends all the processes.

File: report_my_wreck/report_my_wreck/MapView.swift (hunk 1 above)
Comment (JustinW1395) [reply report_my_wreck/report_my_wreck/MapView.swift:17]: .isDetailLink basically sets the navigation link to present its destination as it's detail component which here we want it to be false. It's also on the contentview as well.

**Review by RJB888 — APPROVED** [review]

**Comment by RJB888** [comment]
Looks good.

**Pull request merged**